endpoint.cpp

```diff
@@ -344,6 +344,48 @@ void Endpoint::print_statistics()
     printf("\n}\n");
 }
 
+void Endpoint::untrim_msg(const struct buffer *from, struct buffer *to)
```

Comment:
what do you mean with: in the only place you need ?
the payload is where the zeros needs to be written.

I can not move everything after payload and add the missing zeros because it would overlap the next message.